include/circt/Dialect/FIRRTL/FIRRTLCanonicalization.td

```diff
@@ -44,6 +44,9 @@ def NullAttr : Constraint<CPred<"!$0">>;
 // Constraint that enforces equal types
 def EqualTypes : Constraint<CPred<"$0.getType() == $1.getType()">>;
 
+// Constraint that enforces equal type sizes
+def EqualIntSize : Constraint<CPred<"cast<IntType>($0.getType()).getWidth() == cast<IntType>($1.getType()).getWidth()">>;
```

Comment:
This'll report true for uninferred integers (`std::nullopt`) I think, which is fine (and pattern is correct since also checks one of them has known width), but just want to make sure that's the intent.  Seems fine to me.

Reply:
Yea, I don't want to overlap with KnownWidth